parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -2123,6 +2123,9 @@ private PageHeader newDataPageV2Header(
       int dlByteLength) {
     DataPageHeaderV2 dataPageHeaderV2 = new DataPageHeaderV2(
         valueCount, nullCount, rowCount, getEncoding(dataEncoding), dlByteLength, rlByteLength);
+    if (compressedSize == 0) {
+      dataPageHeaderV2.setIs_compressed(false);
```

Comment:
I'm surprised that `dataPageHeaderV2.setIs_compressed()` has never been called before.

Reply:
BTW, I think the description in the spec needs to be improved since it does not consider the case when `comressed_page_size` is 0: https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L665-L668

>  Which means the section of the page between definition_levels_byte_length + repetition_levels_byte_length + 1 and compressed_page_size (included) is compressed with the compression_codec.

Reply:
Well, I don't think `compressed_page_size` can be 0, except if you have 0 levels and 0 data (is that possible?).

Reply:
My bad. Fixed my comment.